effort_controllers/src/joint_group_position_controller.cpp

```diff
@@ -138,12 +138,24 @@ namespace effort_controllers
         // Compute position error
         if (joint_urdfs_[i]->type == urdf::Joint::REVOLUTE)
         {
-         angles::shortest_angular_distance_with_limits(
-            current_position,
-            command_position,
-            joint_urdfs_[i]->limits->lower,
-            joint_urdfs_[i]->limits->upper,
-            error);
+          if(joint_urdfs_[i]->limits->upper - joint_urdfs_[i]->limits->lower >= 2*M_PI)
+          {
+            angles::shortest_angular_distance_with_large_limits(
+              current_position,
+              command_position,
+              joint_urdfs_[i]->limits->lower,
+              joint_urdfs_[i]->limits->upper,
+              error);
+          }
```

Comment:
Is it necessary to perform this check? Since `ros_control` would have several issues if `lower < upper` is violated, I think it is safe to assume that `lower` will always be less than `upper`. As such, wouldn't it be equivalent to just always use the new `shortest_angular_distance_with_large_limits`?

Reply:
No, I don't think it is necessary - I left it since I was not entirely sure if at any point one could have `lower > upper`. I will remove it!